server/src/main/java/org/opensearch/telemetry/TelemetrySettings.java

```diff
@@ -53,7 +67,14 @@ public class TelemetrySettings {
     private volatile boolean tracingEnabled;
     private volatile double samplingProbability;
 
-    public TelemetrySettings(Settings settings, ClusterSettings clusterSettings) {
+    private static Settings settingz;
```

Comment:
Please remove these static references, the checks are settings specific and should not be static